scipy/linalg/decomp_svd.py

```diff
@@ -279,6 +283,98 @@ def diagsvd(s, M, N):
         raise ValueError("Length of s must be M or N.")
 
 
+def hosvd(
+    a: npt.ArrayLike, *, full_tensor: bool = True, check_finite: bool = True
+) -> Tuple[List[numpy.ndarray], numpy.ndarray]:
```

Comment:
```suggestion
) -> Tuple[List[npt.NDArray[np.float64], npt.NDArray[np.float64]]:
```
With the release of [numpy 1.21](https://numpy.org/devdocs/reference/typing.html?highlight=typing#numpy.typing.mypy_plugin.NDArray) we can now express the dtype of arrays.

Reply:
But the dtype of the result may not only be float64 but depend on the input type of the function. Since `svd()` supports complex array, this could be a result as well...

Reply:
You're right, actually. I made a mistake when checking the dtypes.
Feel free to ignore this suggestion.